src/ctype/CTypeUtils.ts

```diff
@@ -40,12 +40,26 @@ export function verifyClaimStructure(claim: any, schema: any): boolean {
 }
 
 export function getHashForSchema(schema: ICType['schema']): string {
-  return Crypto.hashObjectAsStr(schema)
+  const { $id, ...schemaWithoutID } = schema
+  return Crypto.hashObjectAsStr(schemaWithoutID)
+}
+
+export function compileSchema(
+  cType: ICType['schema'],
+  nestedCTypes: Array<ICType['schema']>,
+  claimContents: object
+) {
+  const ajv = new Ajv()
+  ajv.addMetaSchema(CTypeModel)
+  const validated = ajv.addSchema(nestedCTypes).compile(cType)
+  const result = validated(claimContents)
+  return result
 }
```

Comment:
Do you think their needs to be a check, similar to `verifySchemaWithErrors` in `CTypeUtils`?

Reply:
yes

Reply:
Ok, I have updated. 

Let me know if it matches what I would need.